More resource_name cleanup (#2551)
chef/chef-web-docs: layouts/shortcodes/dsl_custom_resource_method_resource_name.md

declare a custom name for that resource. For example:

``` ruby
resource_name :my_resource_name
```

The `resource_name` is only used as a fallback name for display purposes.

> Even for display purposes the name used in recipe code which matches a

Reviewer: ```suggestion
Even for display purposes, the name used in recipe code which matches a
```

`provides` statement is favored over the `resource_name` setting, so the

> `resource_name` has little effect.  In Chef Infra Client 16 and later

Reviewer: ```suggestion
`resource_name` has little effect.  In Chef Infra Client 16 and later,
```

> the first `provides` in a resource declaration also sets the fall-back

Reviewer: ```suggestion
the first `provides` in a resource declaration also sets the fallback
```

> `resource_name`, so it recommended to not set it at all.  The `resource_name`

Reviewer 2: ```suggestion
`resource_name`, so it is recommended to not set it at all.  The `resource_name`
```
